core/tools/generate_names.cc

```diff
@@ -63,6 +63,7 @@ NameDef names[] = {
     {"cvarNameMissing", "@@<cvar-name-missing>"},
     {"ConstantNameMissing", "<ConstantNameMissing>", true},
     {"ErrorNode", "<ErrorNode>", true},
+    {"dynamicConstAssign", "<dynamic-const-assign>"},
```

Comment:
It is a little surprising to me that we don't need to check for this anywhere so it doesn't leak into error messages.

Reply:
Good point. I thought that maybe I could get it to trigger with a pinning error:

```ruby
def foo
  X = false
  1.times do
    X = true
  end
end
```

but it turns out that we don't print the variable name in the pinning error:

```
pin.rb:4: dynamic constant assignment https://srb.help/2001
     4 |  X = false
          ^

pin.rb:6: dynamic constant assignment https://srb.help/2001
     6 |    X = true
            ^

pin.rb:6: Changing the type of a variable in a loop is not permitted https://srb.help/7001
     6 |    X = true
                ^^^^
  Existing variable has type: FalseClass
  Attempting to change type to: TrueClass

  Autocorrect: Use `-a` to autocorrect
    pin.rb:4: Replace with T.let(false, T::Boolean)
     4 |  X = false
              ^^^^^
Errors: 3
```

I will think about whether we need to silence this somewhere.

Reply:
I think the answer is that it's already quite rare for us to print variable names, because CFG creates so many nonsensical variable names (`<statTemp>$3>` etc.)

Reply:
That makes a certain amount of sense; I guess since we're generating the error from the parser and not from someplace like resolver, `dynamicConstAssign` is basically a variable for the rest of the pipeline.